ocs_ci/ocs/platform_nodes.py

```diff
@@ -394,24 +414,64 @@ def start_nodes(self, nodes, wait=True):
         )
         self.aws.start_ec2_instances(instances=instances, wait=wait)
 
-    def restart_nodes(self, nodes, wait=True, force=True):
+    def restart_nodes(self, nodes, wait=True):
         """
         Restart EC2 instances
 
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True if need to wait till the restarted node reaches
+                READY state. False otherwise
+
+        """
+        instances = self.get_ec2_instances(nodes)
+        assert instances, (
+            f"Failed to get the EC2 instances for nodes {[n.name for n in nodes]}"
+        )
+        self.aws.restart_ec2_instances(instances=instances)
+        if wait:
+            """
+            When reboot is initiated on an instance from the AWS, the instance 
+            stays at "Running" state throughout the reboot operation.
+
+            Once the OCP node detects that the node is not reachable then the
+            node reaches status NotReady.
+            When the reboot operation is completed and the instance is up and 
+            reachable the OCP node reaches status Ready.
+            """
+            nodes_names = [n.name for n in nodes]
+            logger.info(
+                f"Waiting for nodes: {nodes_names} to reach not ready state"
+            )
+            wait_for_nodes_status(
```

Comment:
1) When a node is restart from AWS/Vmware the instance/VM stays at running state throughout the reboot operation and I think due to this there is no return wait conditions (wait_until/WaitForTasks). 
2) When we initiated reboot it will take few mins for the node to reach NotReady state and once the instance is reachable then the node reaches Ready state.

When we call restart_nodes() and immediately check for wait_for_nodes_status=Ready it will succeed but actually the OCP node has not detect the failure of node yet, after few mins the OCP node will detect the failure and reach NotReady state, which is misleading.

So, I am first waiting for the node to reach NotReady and then Ready.
I tried it even manually and I see always the node reaching NotReady state first and then Ready state.

Please suggest if there is any other better way/solution to handle this.